server/src/test/scala/services/crunch/ArrivalsGraphStageSpec.scala

```diff
@@ -156,6 +156,63 @@ class ArrivalsGraphStageSpec extends CrunchTestLike {
     }
   }
 
+  "Given a live arrival and a cirium arrival" >> {
+    "When they have matching number, schedule, terminal and origin" >> {
+      "I should see the live arrival with the cirium arrival's status merged" >> {
+        val scheduled = "2021-06-01T12:00"
+        val liveArrival = ArrivalGenerator.arrival("AA0001", schDt = scheduled, terminal = T1, origin = PortCode("AAA"))
+        val ciriumArrival = ArrivalGenerator.arrival("AA0001", schDt = scheduled, terminal = T1, origin = PortCode("AAA"), estDt = scheduled)
+
+        val crunch: CrunchGraphInputsAndProbes = runCrunchGraph(TestConfig(now = () => SDate(scheduled)))
+
+        offerAndWait(crunch.liveArrivalsInput, ArrivalsFeedSuccess(Flights(List(liveArrival))))
+
+        crunch.portStateTestProbe.fishForMessage(1 second) {
+          case PortState(flights, _, _) => flights.nonEmpty
+        }
+
+        offerAndWait(crunch.ciriumArrivalsInput, ArrivalsFeedSuccess(Flights(List(ciriumArrival))))
+
+        crunch.portStateTestProbe.fishForMessage(1 second) {
+          case PortState(flights, _, _) => flights.values.exists(_.apiFlight.Estimated == Option(SDate(scheduled).millisSinceEpoch))
+        }
+
+        success
+      }
+    }
+
+    "When they have matching number, schedule, terminal and origin" >> {
+      "I should see the live arrival without the cirium arrival's status merged" >> {
+        val scheduled = "2021-06-01T12:00"
+        val liveArrival = ArrivalGenerator.arrival("AA0002", schDt = scheduled, terminal = T1, origin = PortCode("AAA"))
+        val ciriumArrival = ArrivalGenerator.arrival("AA0002", schDt = scheduled, terminal = T1, origin = PortCode("BBB"), estDt = scheduled)
+
+        val crunch: CrunchGraphInputsAndProbes = runCrunchGraph(TestConfig(now = () => SDate(scheduled)))
+
+        offerAndWait(crunch.liveArrivalsInput, ArrivalsFeedSuccess(Flights(List(liveArrival))))
+
+        crunch.portStateTestProbe.fishForMessage(1 second) {
+          case PortState(flights, _, _) => flights.nonEmpty
+        }
+
+        offerAndWait(crunch.ciriumArrivalsInput, ArrivalsFeedSuccess(Flights(List(ciriumArrival))))
+
+        Thread.sleep(500)
```

Comment:
do we definitely need this sleep?

Reply:
Yes to make sure enough time has passed for the cirium arrival to have been processed and ignored